libp2p.nim

```diff
@@ -8,20 +8,26 @@
 ## those terms.
 
 import
-  libp2p/[daemon/daemonapi,
-          daemon/transpool,
-          protobuf/minprotobuf,
+  libp2p/[protobuf/minprotobuf,
+          muxers/muxer,
+          muxers/mplex/mplex,
+          stream/lpstream,
+          stream/bufferstream,
+          stream/connection,
+          transports/transport,
+          transports/tcptransport,
+          protocols/secure/noise,
           varint,
           switch,
           peerid,
           peerinfo,
-          stream/lpstream,
-          stream/bufferstream,
-          stream/connection,
           multiaddress,
           crypto/crypto]
 
+import bearssl
+
 export
-  daemonapi, transpool, minprotobuf, varint,
-  switch, peerid, peerinfo, connection,
-  multiaddress, crypto, lpstream, bufferstream
+  minprotobuf, varint,switch, peerid, peerinfo,
+  connection, multiaddress, crypto, lpstream,
+  bufferstream, bearssl, muxer, mplex, transport,
```

Comment:
not sure internals like bearssl and varint necessarily should be exported, they're not part of the public api exposed in this module and pollute the global namespace with many ambiguous symbols - the rule is basically "anything that forms part of the public api should be exported"

Reply:
I'm fine with `bearssl`, but though I expose it since it's now used as the default RNG provider. 

Varint seems like a good candidate to export tho, since it's used extensively across libp2p and can be useful for custom parsers and such?

Reply:
> I'm fine with bearssl, but though I expose it since it's now used as the default RNG provider.

ok - if any  bear type appears in public API needed by this module, then yes

> Varint seems like a good candidate to export tho,

depends - ie libp2p consumes varint but presents a more sane API to to user that doesn't expose any varints, so I would tend towards no, for pollution reasons (ie if the user is using another varint library, of which there are plenty, they'll get into symbol trouble which is annoying and easily causes breakage

Reply:
> depends - ie libp2p consumes varint but presents a more sane API to to user that doesn't expose any varints, so I would tend towards no, for pollution reasons (ie if the user is using another varint library, of which there are plenty, they'll get into symbol trouble which is annoying and easily causes breakage

OK, I don't disagree since it can be imported implicitly.

Reply:
> ok - if any bear type appears in public API needed by this module, then yes

Yeah, we need `BrHmacDrbgContext` which seems to be part of bearssl, is returned by `newRng()`.